tikv-client/src/main/java/com/pingcap/tikv/key/IndexKey.java

```diff
@@ -49,8 +49,10 @@ public static IndexKey toIndexKey(long tableId, long indexId, Key... dataKeys) {
     for (int i = 0; i < indexColumns.size(); i++) {
       TiIndexColumn col = indexColumns.get(i);
       DataType colTp = tableInfo.getColumn(col.getOffset()).getType();
+      // using convertToTiDBType implicitly since the get method does not convert
+      Object convertedValue = colTp.convertToTiDBType(row.get(col.getOffset(), colTp));
```

Comment:
I feel like conversion should be done at the upper level. At this stage, row is already converted. Please correct me if my understanding is not quite right.

Reply:
updated